### PR TITLE
Migrate console

### PR DESCRIPTION
**Pull request by siddhsuresh**
Closes: #3862
Closes: #3858
Closes: https://github.com/blitz-js/blitz/issues/1098

### What are the changes and their implications?

- [x] migrate required files from legacy framework repo 
- https://github.com/blitz-js/legacy-framework/blob/main/nextjs/packages/next/cli/next-console.ts
- https://github.com/blitz-js/legacy-framework/blob/main/packages/cli/src/commands/console.ts
- [x] use ```esbuild``` instead of ```ts-node``` with reference to https://github.com/blitz-js/blitz/issues/1098#issuecomment-823670547 
- [x] add ```--only-db``` flag allows users to only load the ```db``` module in the initial load of the console but allows usage other modules when changes are made 

> In the following images the "Loaded ... with the time" is only for reference and is not present in the code

### Before esbuild

image

### After esbuild

image

### With ```skip-preload```
image

### If the ```db``` folder does not exist by the user provide the location of ```prisma``` schema
```js
//package.json

"prisma": {
    "schema": "prisma/schema.prisma"
  },
```
image

## Only reloading the modules that are changed and not all when the user makes changes

```ts
const setupFileWatchers = async (repl: REPLServer) => {
  debug("Setting up file watchers...")
  const watchers = [
    watch(await getBlitzModulePaths(), {
      ignoreInitial: true,
    }).on(
      "all",
      async (event:string, path: string) => { 👈🏼
        const modulePath = path
        const modules = await loadBlitz(false, modulePath) 👈🏼
        loadBlitzModules(repl, modules)
      }
    )
  ]
```

For example when a user changes the db/index.ts file only that file is loaded by the code 

image

> pnpm blitz -v
Blitz version: 2.0.0-beta.4 (local)
Windows 11 | win32-x64 | Node: v16.13.0
 Package manager: npm
  System:
    OS: Windows 10 10.0.22621
    CPU: (4) x64 11th Gen Intel(R) Core(TM) i3-1115G4 @ 3.00GHz
    Memory: 570.32 MB / 7.70 GB
  Binaries:
    Node: 16.13.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.17 - ~\AppData\Roaming\npm\yarn.CMD
    npm: 8.1.0 - C:\Program Files\nodejs\npm.CMD

### CODE REVIEW

**Review by beerose — CHANGES_REQUESTED** [review]
This is amazing work! Thank you!

I started testing it and it looks like top-level await doesn't work:

CleanShot 2022-09-26 at 13 31 40@2x

**Comment by siddhsuresh** [comment]
> This is amazing work! Thank you!
> 
> I started testing it and it looks like top-level await doesn't work:
> 
> CleanShot 2022-09-26 at 13 31 40@2x

Hey @beerose, I am not getting that error on my end. I just tested in ubuntu through WSL2 and here is the terminal output. (Ignore the speed of my terminal, WSL2 in my laptop is very slow 😅)

[asciicast](https://asciinema.org/a/rrMs9Lb6xl51GB68am0mRtETk)

**Comment by beerose** [comment]
Hmm, that's interesting! I wouldn't expect it to be OS related, but maybe it is? I'm on macos, Node 16.

**Comment by siddhsuresh** [comment]
> Hmm, that's interesting! I wouldn't expect it to be OS related, but maybe it is? I'm on macos, Node 16.

@beerose I am not sure what is going wrong because it works for me on both windows 11 and in ubuntu

**Comment by siddhsuresh** [comment]
> This is amazing work! Thank you!
> 
> I started testing it and it looks like top-level await doesn't work:
> 
> CleanShot 2022-09-26 at 13 31 40@2x

@beerose After researching how the REPL works with top level await, I have introduced this fix.
Now users can write commands without writing the ```await``` as I have overridden the eval code of the NodeJS REPL to resolve all commands as promises

### Before
image

### After 
image

Does this solve the problem on your end?

**Comment by beerose** [comment]
@siddhsuresh we should also change the message:

```
Tips: - Exit by typing .exit or pressing Ctrl-D
      - Use your db like this: await db.project.findMany()
      - ...
```

(remove the await)

**Comment by beerose** [comment]
@siddhsuresh I was also thinking about the `skip-preload` and I'm not sure it's obvious what it means (if someone didn't read the docs first). Maybe we can call thins flag `--skip-resolvers`?

**Comment by siddhsuresh** [comment]
> @siddhsuresh I was also thinking about the `skip-preload` and I'm not sure it's obvious what it means (if someone didn't read the docs first). Maybe we can call thins flag `--skip-resolvers`?

@beerose yeah I agree, I had given the flag name by taking reference from #3858, but also was thinking of changing it to ```only-db``` because it is exactly what the flag would do. What are your thoughts on this? 
If this would not be suitable I agree with naming it ```skip-resolvers```

**Comment by beerose** [comment]
> > @siddhsuresh I was also thinking about the `skip-preload` and I'm not sure it's obvious what it means (if someone didn't read the docs first). Maybe we can call thins flag `--skip-resolvers`?
> 
> @beerose yeah I agree, I had given the flag name by taking reference from #3858, but I was also thinking of changing it to `--only-db` because it is exactly what the flag would do. What are your thoughts on this? If this would not be suitable I agree with naming it `skip-resolvers`

I was also thinking about `only-db` or `db-only`, so totally up to you :)

**Comment by siddhsuresh** [comment]
> > > @siddhsuresh I was also thinking about the `skip-preload` and I'm not sure it's obvious what it means (if someone didn't read the docs first). Maybe we can call thins flag `--skip-resolvers`?
> > 
> > 
> > @beerose yeah I agree, I had given the flag name by taking reference from #3858, but I was also thinking of changing it to `--only-db` because it is exactly what the flag would do. What are your thoughts on this? If this would not be suitable I agree with naming it `skip-resolvers`
> 
> I was also thinking about `only-db` or `db-only`, so totally up to you :)

@beerose I have changed the flag name to ```--only-db```

File: .changeset/spotty-zoos-film.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"blitz": patch
+---
+
+Migrate console
```
Comment (beerose) [inline .changeset/spotty-zoos-film.md:5]: Add `blitz console` CLI command back to toolkit

**Review by beerose — APPROVED** [review]
Amazing work!

**Review by dillonraphael — APPROVED** [review]

**Pull request merged**